analyzers/tests/SonarAnalyzer.UnitTest/Extensions/DiagnosticDescriptorExtensionsTest.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Extensions;
+
+using static SonarAnalyzer.Helpers.DiagnosticDescriptorFactory;
+
+namespace SonarAnalyzer.UnitTest.Extensions;
+
+[TestClass]
+public class DiagnosticDescriptorExtensionsTest
+{
+    private const string MainTag = "MainSourceScope";
+    private const string TestTag = "TestSourceScope";
+    private const string UtilityTag = "Utility";
+
+    [TestMethod]
+    public void HasMatchingScope_NoCompilation_IsMatching() =>
+        TestHelper.CreateDescriptor("Sxxx").HasMatchingScope(null, true, false).Should().BeTrue();
+
+    [DataTestMethod]
+    [DataRow(true, ProjectType.Product, MainTag)]
+    [DataRow(true, ProjectType.Product, MainTag, UtilityTag)]
+    [DataRow(true, ProjectType.Product, MainTag, TestTag)]
+    [DataRow(true, ProjectType.Product, MainTag, TestTag, UtilityTag)]
+    [DataRow(true, ProjectType.Test, TestTag)]
+    [DataRow(true, ProjectType.Test, TestTag, UtilityTag)]
+    [DataRow(true, ProjectType.Test, MainTag, TestTag)]
+    [DataRow(true, ProjectType.Test, MainTag, TestTag, UtilityTag)]
+    [DataRow(false, ProjectType.Product, TestTag)]
+    [DataRow(false, ProjectType.Product, TestTag, TestTag)]
+    [DataRow(false, ProjectType.Test, MainTag)]
+    [DataRow(false, ProjectType.Test, MainTag, MainTag)]
+    public void HasMatchingScope_SingleDiagnostic_WithOneOrMoreScopes_SonarLint(bool expectedResult, ProjectType projectType, params string[] ruleTags)
+    {
+        var compilation = new SnippetCompiler("// Nothing to see here").SemanticModel.Compilation;
+        var diagnostic = TestHelper.CreateDescriptor("Sxxx", ruleTags);
```

Comment:
I would extract `"Sxxx"` to a constant and change it to be similar with our rule ids (e.g. `S0001`).

Reply:
I've extracted the constant.
I want to keep `xxx` part, as `S` prefix makes it similar with our rule IDs and `xxx` makes clear that it is none of our rules and no additional scaffolding (like rspec) is needed for this part to work.